### PR TITLE
adds ironic conductor liveness check

### PR DESCRIPTION
**Pull request by stefanhipfel**

### CODE REVIEW

**Review by sandzwerg — APPROVED** [review]

File: agentliveness/agent.py [hunk 1]
```diff
@@ -175,6 +179,38 @@ def _check_manila(self):
...
+                return 1
+            
+            drivers = ironic.driver.list()
+            ipmi = next(
+                (driver for driver in drivers if driver.name == 'ipmi'),
+                None
+            )
+            if ipmi is None or self.CONF.ironic_conductor_host not in ipmi.hosts:
+                logger.error("Conductor %s is not listed in ipmi driver, commencing suicide", self.CONF.ironic_conductor_host)
+                return 1
```
Comment (fwiesel) [inline agentliveness/agent.py:206]: Why specifically `ipmi` and not any of the other drivers?
I hope, we can get rid of ipmi rather earlier than later.

I.e.:
```python

for driver in drivers:
   if self.CONF.ironic_conductor_host in driver.hosts:
      break
else:
  logger.error("Conductor %s is not listed for any driver, commencing suicide", self.CONF.ironic_conductor_host)
  return 1
```

**Review by rhalina — APPROVED** [review]
LGTM

File: agentliveness/agent.py (hunk 1 above)
Comment (sandzwerg) [reply agentliveness/agent.py:206]: At that time only IPMI was used but I agree

Comment (fwiesel) [reply agentliveness/agent.py:206]: Okay, I'll create a follow up PR.

Comment (fwiesel) [reply agentliveness/agent.py:206]: Follow up is #6

**Review by fwiesel — APPROVED** [review]

**Pull request merged**